sdk/go/pulumi/types_test.go

```diff
@@ -339,6 +339,11 @@ func TestToOutputAnyDeps(t *testing.T) {
 	}()
 
 	res := &ResourceState{}
+	urnOut := URNOutput{newOutputState(nil, reflect.TypeOf(URN("")), res)}
+	go func() {
```

Comment:
Why's this in a goroutine?

Reply:
@iwahbe this is my question as well - does `urnOut.resolve()` have side effects that occur in the same thread?

Reply:
Putting this in a goroutine probabilistically tests that the code is able to properly await unresolved outputs.